internal/clients/http/rbac/rbac_client.go

```diff
@@ -0,0 +1,106 @@
+	"github.com/RHEnVision/provisioning-backend/internal/clients/http"
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/headers"
+	"github.com/RHEnVision/provisioning-backend/internal/metrics"
+	"github.com/RHEnVision/provisioning-backend/internal/ptr"
+	"github.com/rs/zerolog"
+	"go.opentelemetry.io/otel"
+)
+
+const TraceName = "github.com/EnVision/provisioning/internal/clients/http/rbac"
+
+type rbac struct {
+	client *ClientWithResponses
+}
+
+func init() {
+	clients.GetRbacClient = newClient
+}
+
+func logger(ctx context.Context) zerolog.Logger {
+	return zerolog.Ctx(ctx).With().Str("client", "ib").Logger()
+}
+
+func newClient(ctx context.Context) clients.Rbac {
+	if !config.Application.RbacEnabled {
+		return &allPermRbacSingleton
+	}
+
+	c, err := NewClientWithResponses(config.RBAC.URL, func(c *Client) error {
+		c.Client = http.NewPlatformClient(ctx, config.ImageBuilder.Proxy.URL)
+		return nil
+	})
+	if err != nil {
+		// in case there was an error during initialization return "no permissions ACL"
+		zerolog.Ctx(ctx).Warn().Err(err).Msg("Could not initialize RBAC client, returning empty ACL")
+		return &noPermRbacSingleton
+	}
+	return &rbac{client: c}
+}
+
+func (c *rbac) Ready(ctx context.Context) error {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "Ready")
+	defer span.End()
+
+	logger := logger(ctx)
+	resp, err := c.client.GetStatus(ctx, headers.AddRbacIdentityHeader, headers.AddEdgeRequestIdHeader)
+	if err != nil {
+		logger.Error().Err(err).Msgf("Readiness request failed for RBAC: %s", err.Error())
+		return err
+	}
+	defer resp.Body.Close()
+
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode)
+	if err != nil {
+		return fmt.Errorf("ready call: %w", err)
+	}
+	return nil
+}
+
+// ErrPaginationUnsupported is returned instead of paginated payload. This is unsupported in the official rbac-client-go.
+var ErrPaginationUnsupported = errors.New("RBAC returned paginated result")
+
+func (c *rbac) GetPrincipalAccess(ctx context.Context) (clients.RbacAcl, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "GetPrincipalAccess")
+	defer span.End()
+
+	start := time.Now()
+	defer func() {
+		metrics.RbacAclFetchDuration.Observe(float64(time.Since(start).Nanoseconds()) / 1000000)
+	}()
+
+	logger := logger(ctx)
+	params := GetPrincipalAccessParams{
+		Application: "provisioning",
+		Limit:       ptr.To(QueryLimit(999)),
+	}
+	resp, err := c.client.GetPrincipalAccessWithResponse(ctx, &params, headers.AddRbacIdentityHeader, headers.AddEdgeRequestIdHeader)
+	if err != nil {
+		logger.Error().Err(err).Msgf("Get principal call failed: %s", err.Error())
+		return nil, err
+	}
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
+	if err != nil {
+		return nil, fmt.Errorf("get principal: %w", err)
+	}
+
+	if resp.JSON200.Meta.Count != nil && *resp.JSON200.Meta.Count > 1 {
+		return nil, ErrPaginationUnsupported
+	}
+
```

Comment:
This behavior is documented few lines above: `// ErrPaginationUnsupported is returned instead of paginated payload. This is unsupported in the official rbac-client-go.`

I was simply lazy, it is unlikely we will ever get more permissions than 999, the hardcoded limit. But let me do pagination it is simple. Rebased.